entity-service-impl/src/main/java/org/hypertrace/entity/service/util/DocStoreConverter.java

```diff
@@ -183,6 +185,28 @@ private static Filter transformToOrFilterChainForStrArray(AttributeFilter attrib
     return f;
   }
 
+  private static Filter transformToAndFilterChainForStrArray(AttributeFilter attributeFilter) {
+    String fieldName = attributeFilter.getName() + VALUE_LIST_VALUES_CONST;
+
+    Filter f = new Filter();
+    f.setFieldName("");
+    f.setOp(Op.AND);
+
+    Filter exists = new Filter();
+    exists.setFieldName(fieldName);
+    exists.setOp(Op.EXISTS);
```

Comment:
I am not exactly sure whether we need Exists or Not Exists here. Won't we want to return all the entities which do not have this field?

Here we are doing `labels NOT IN [A, B]`
- Do we only want to return entities which has labels but does not have A or B
- Or, do we also want to return entities where labels does not exist as well?

Reply:
I would think #2.

Reply:
@kotharironak In that case, we need to do "NOT_EXISTS OR the other filter"

Reply:
@skjindal93 Can you check this integration test - `testNotInFilterForMatchingLabelsQuery`? It has `5` documents. One doesn't have `attributes.labels` fields. Is it expected in the result?

Reply:
> Can you check this integration test - testNotInFilterForMatchingLabelsQuery? It has 5 documents. One doesn't have attributes.labels fields. Is it expected in the result?

Yeah, `createdEntity5` is expected in the result

Reply:
@skjindal93 I have added some more integration tests for `NOT_IN`, `IN`, and `EQ` - `testDifferentFilterForMatchingLabelsQuery`

Reply:
Yes @kotharironak That entity is expected in the result

Reply:
Made the changes. @avinashkolluru @skjindal93, now it's good to go. Have a look.